### PR TITLE
wip: [example-browser] Change backend startup method

### PR DESCRIPTION
**Pull request by marechal-p**
Before, we were executing the server in the process that used to run the tests.

Changed that so that the backend is fully independent from the test process.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

### CODE REVIEW

**Comment by marechal-p** [comment]
@marcdumais-work can you tell me if this improves stability for you? Every time I think it might be improving, I get another random failure. The flakiness is killing me.

cc @simark, if you have any opinion regarding this change.

File: packages/core/src/node/messaging/messaging-contribution.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ export class MessagingContribution implements BackendApplicationContribution, Me
 
     protected createChannel(id: number, socket: ws): WebSocketChannel {
         return new WebSocketChannel(id, content => {
-            if (socket.readyState < ws.CLOSING) {
+            if (socket.readyState === ws.OPEN) {
```
Comment (akosyakov) [inline packages/core/src/node/messaging/messaging-contribution.ts]: Can you comment on what was the issue?

**Comment by marechal-p** [comment]
@akosyakov Can you give me your opinion on these changes?

I know it is better to not fix something that doesn't look broken, but I remember having issues. So I tried to start the backend in a more "clean" way, isolated in its own separate process, in a much more similar way than what we would actually do.

Also catches the coverage failure at then end, because it seems like for most tests we don't generate anything anyway.

File: packages/core/src/node/messaging/messaging-contribution.ts (hunk 1 above)
Comment (akosyakov) [reply packages/core/src/node/messaging/messaging-contribution.ts]: ok, let's give it a try

**Comment by akosyakov** [comment]
@marechal-p sorry, i don't want to dive into it, i would be fine if @simark, @marcdumais-work review it

File: examples/browser/test/main-page/main-page.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class MainPage {
     }
 
     waitForStartup(): void {
-        this.driver.waitUntil(() => !this.driver.isExisting('.theia-preload'));
+        this.driver.waitUntil(() => this.driver.isExisting('.theia-preload'));
```
[the remark this answers is not in the document]
Comment (vince-fugnitto) [reply examples/browser/test/main-page/main-page.ts]: another option is to simply call 
```ts
this.driver.waitForExist('.theia-preload', 30000, true);
```
where `true` specifies that we want the reverse (wait for it to not exist)

[the remark this answers is not in the document]
Comment (marechal-p) [reply examples/browser/test/main-page/main-page.ts]: It was part of some tests, I had forgot to remove it, nice catch.

**Comment by marechal-p** [comment]
https://github.com/theia-ide/theia/pull/4089 will fix the issues with AppVeyor :tada:

**Pull request closed without merging**